doc/cabal-commands.rst

```diff
@@ -142,6 +182,57 @@ arguments and flags that are common to some or all of them.
     Already generated `build-info.json` files will be removed since they would be stale otherwise.
 
 
+cabal list-bin
+--------------
+
+``cabal list-bin`` will either; (a) display the path for a single exe or (b)
+complain that the target doesn't resolve to a single binary. It won't list
+multiple binary paths but knows which build products a package has.
```

Comment:
How about:
```suggestion
complain that the target doesn't resolve to a single binary. In the latter case,
it will name the binary products contained in the package. These products can 
be used to narrow the search and get an actual path to a particular executable.

Example showing a failure to resolve to a single executable.
```